python/cuml/dask/datasets/regression.py

```diff
@@ -179,59 +274,99 @@ def make_regression(n_samples=100, n_features=100, n_informative=10,
         The coefficient of the underlying linear model. It is returned only if
         coef is True.
     """
+
+    client = default_client() if client is None else client
+
     n_informative = min(n_features, n_informative)
     rs = create_rs_generator(random_state)
 
     if n_samples_per_part is None:
         n_samples_per_part = max(1, int(n_samples / n_parts))
 
-    if effective_rank is None:
+    if (effective_rank is None) or (effective_rank and not use_full_low_rank):
         # Randomly generate a well conditioned input set
-        X = rs.standard_normal((n_samples, n_features),
-                               chunks=(n_samples_per_part, (n_informative,
-                                                            n_features -
-                                                            n_informative)))
+        if order == 'F':
+            X = f_order_standard_normal(client, rs, n_samples, n_features,
+                                        [n_samples_per_part] * n_parts, dtype)
+        elif order == 'C':
+            X = rs.standard_normal((n_samples, n_features),
+                                   chunks=(n_samples_per_part, (
+                                                                n_informative,
+                                                                n_features -
+                                                                n_informative)
+                                           ),
+                                   dtype=dtype)
```

Comment:
If I well understood the problem discussed with @cjnolet, the C++ side of the code reads the matrix in the wrong order. I think `X` should be reordered for both normal and low rank matrices. `X` is just made out of random values, so the main point is that the right matrix elements are used during the computation of y so that the regression makes sense on the C++ side. Just a suggestion, but if it is possible to modify strides of each chunk in a Dask array, it could be possible to just generate a normal array with Dask random generator and then reorder `X` inplace. That also provides a solution for reordering the low rank matrix.

I think ground_truth is in 'C' order by default too, so it should normally be reordered too before return. This probably can't be done inplace (as not random), but this is a smaller matrix. If `X` and `ground_truth` are in the right order `y` should be as well (might need a verification though).

Reply:
@wxbn The random numbers that I generate do exactly that. I create a 1d random standard normal, and reshape it in the correct shape and order. That is in-place, without a copy, and merely updates the strides and the flags. As for ground truth and y, I will do the required updates to order. I think they are only required when `n_targets > 1`

Reply:
Also, even if `ground_truth` and `y` are in `C` order for `n_targets > 1`, they will be converted by cuml to the correct order

Reply:
New commit transposes `ground_truth` and `y` explicity for `n_targets > 3`

Reply:
> X is just made out of random values, so the main point is that the right matrix elements are used during the computation of y so that the regression makes sense on the C++ side.

@wxbn, @divyegala just brought to my attention that Scikit-learn indeed does not use the low-rank matrix to generate the coefficients. Indeed, I should have looked more closely because the documentation and the Dask implementation gave me the impression that the returned coefficients were in fact the solution to Ax=b. The little hack that the Dask version is doing still works because the linear relationship is preserved between X and y, it just ignores the generated low-rank.

We can still generate the data from the low-rank computed on a single partition, however, by computing the covariance from the low rank and passing that, along with the generated coefficients, to the other partitions so they can generate similar feature distributions and maintain the same linear relationship between X and y across the cluster.